keras_cv/models/segmentation/deeplab_v3/deeplab_v3.py

```diff
@@ -0,0 +1,260 @@
+        spatial_pyramid_pooling: (Optional) a `keras.layers.Layer`. Also known as
+            Atrous Spatial Pyramid Pooling (ASPP). Performs spatial pooling on
+            different spatial levels in the pyramid, with dilation. If provided,
+            the feature map from the backbone is passed to it inside the DeepLabV3
+            Encoder, otherwise `keras_cv.layers.spatial_pyramid.SpatialPyramidPooling`
+            is used.
+        low_level_feature_layer_name: (Optional) str, which refers to the name of the
+            intermediate layer from the `backbone` from which the low-level features
+            will be extracted and combined with the encoder features to be passed to
+            the `segmentation_head`. If not specified, no low-level features from the
+            `backbone` will be combined with the encoder output and the resulting
+            architecture is that of a DeepLabV3 model, otherwise a DeepLabV3+ model.
+        projection_filters: (Optional) int, number of filters in the convolution layer
+            projecting low-level features from the `backbone`. This parameter is only
+            relevant if `low_level_feature_layer_name` is also specified.
+        segmentation_head: (Optional) a `keras.layers.Layer`. If provided, the
+            outputs of the DeepLabV3 encoder is passed to this layer and it should
+            predict the segmentation mask based on feature from backbone and feature
+            from decoder, otherwise a similar architecture is used.
+        segmentation_head_activation: Optional `str` or function, activation
+            functions between the `keras.layers.Conv2D` layers and the final
+            classification layer, defaults to `"relu"`.
+        input_shape: optional shape tuple, defaults to `(None, None, 3)`.
+        input_tensor: optional Keras tensor (i.e., output of `layers.Input()`)
+            to use as image input for the model.
+
+    Examples:
+    ```python
+    import tensorflow as tf
+    import keras_cv
+
+    images = tf.ones(shape=(1, 96, 96, 3))
+    labels = tf.zeros(shape=(1, 96, 96, 1))
+    backbone = keras_cv.models.ResNet50V2Backbone(input_shape=[96, 96, 3])
+    model = keras_cv.model.segmentation.DeepLabV3(
+        num_classes=1,
+        backbone=backbone,
+        low_level_feature_layer_name="v2_stack_0_block1_2_relu"
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Train model
+    model.compile(
+        optimizer="adam",
+        loss=keras.losses.BinaryCrossentropy(from_logits=False),
+        metrics=["accuracy"],
+    )
+    model.fit(images, labels, epochs=3)
+    ```
+    """
+
+    def __init__(
+        self,
+        num_classes,
+        backbone,
+        spatial_pyramid_pooling=None,
+        low_level_feature_layer_name=None,
+        projection_filters=None,
+        segmentation_head=None,
+        segmentation_head_activation="softmax",
+        dropout=0.0,
+        input_shape=(None, None, 3),
```

Comment:
Input shape and input_tensor can/should be handled by the backbone, so we should be able to drop them as args here